engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/AtomConstructor.java

```diff
@@ -90,13 +93,39 @@ public AtomConstructor(String name, Module definitionModule, Type type, boolean
    * @return {@code true} if {@link #initializeFields} method has already been called
    */
   public boolean isInitialized() {
-    return constructorFunction != null;
+    return accessor != null;
   }
 
   boolean isBuiltin() {
     return builtin;
   }
 
+  /**
+   * Building blocks required for initialization of the atom constructor.
+   *
+   * @param section the source section
+   * @param localScope the description of the local scope
+   * @param assignments the expressions that evaluate and assign constructor arguments to local vars
+   * @param varReads the expressions that read field values from local vars
+   * @param annotations the list of attached annotations
+   * @param args the list of argument definitions
+   */
+  public record InitializationParts(
```

Comment:
Does this new record have to be public? Because the `record` defines not just public constructor, but also public getters. I do not think we want people around the code base to call the getters.

https://github.com/enso-org/enso/blob/0c41e8d802e4a4823c73d7c4d4d3a621c81ab27d/engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/function/FunctionSchema.java#L235

Could we rather follow this example by @Akirathan where Pavel introduced `FunctionSchema.Builder`?